File: test-suite-python/src/test/python/micronaut/views/docs/views/ViewsController.py
from micronaut.context.annotation import Requires
from micronaut.http import HttpResponse
from micronaut.http.annotation import Controller, Get
from micronaut.views import ModelAndView, View

from .Person import Person


@Requires(property="spec.name", value="ViewsControllerTest")
# tag::clazz[]
@Controller("/views")
class ViewsController:
# end::clazz[]

    # tag::map[]
    @View("home")  # <1>
    @Get
    def index(self) -> HttpResponse:
        return HttpResponse.ok({"loggedIn": True, "username": "sdelamo"})
    # end::map[]

    # tag::pojo[]
    @View("home")  # <1>
    @Get("/pojo")
    def pojo(self) -> HttpResponse:
        return HttpResponse.ok(Person("sdelamo", True))
    # end::pojo[]

    # tag::modelAndView[]
    @Get("/modelAndView")
    def model_and_view(self) -> ModelAndView:
        return ModelAndView("home", Person("sdelamo", True))
    # end::modelAndView[]
# tag::endclass[]
# end::endclass[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/views/docs/views/ViewsController.kt
package io.micronaut.views.docs.views

import io.micronaut.context.annotation.Requires
import io.micronaut.http.HttpResponse
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.views.ModelAndView
import io.micronaut.views.View

@Requires(property = "spec.name", value = "ViewsControllerTest")
//tag::clazz[]
@Controller("/views")
class ViewsController {
//end::clazz[]

    //tag::map[]
    @View("home") // <1>
    @Get
    fun index(): HttpResponse<*> {
        return HttpResponse.ok(mapOf("loggedIn" to true, "username" to "sdelamo"))
    }
    //end::map[]

    //tag::pojo[]
    @View("home") // <1>
    @Get("/pojo")
    fun pojo(): HttpResponse<Person> {
        return HttpResponse.ok(Person("sdelamo", true))
    }
    //end::pojo[]

    //tag::modelAndView[]
    @Get("/modelAndView")
    fun modelAndView(): ModelAndView<Person> {
        return ModelAndView("home", Person("sdelamo", true))
    }
    //end::modelAndView[]
//tag::endclass[]
}
//end::endclass[]
